firebase-firestore/src/main/java/com/google/firebase/firestore/Query.java

```diff
@@ -84,6 +84,77 @@ public FirebaseFirestore getFirestore() {
     return firestore;
   }
 
+  /** Validates the given document ID and returns the corresponding FieldValue. */
+  private FieldValue validateAndCalculateReferenceValue(FieldPath fieldPath, Object value) {
```

Comment:
I think I would:
1. Rename this to parseDocumentIdValue().  "parse" is a good verb to use when we're both validating and transforming an input.
2. Rename "value" to "documentIdValue"
3. Change the return type to ReferenceValue which is more specific than FieldValue so makes the code more type-safe and self-documenting.
4. Change the comment to something like "Parses the given documentIdValue into a ReferenceValue, throwing appropriate errors if the value is anything other than a DocumentReference or String or if the string is malformed."